Teach the explicit constructor check about constexpr.

danakj opened the pull request:
If a constructor is marked constexpr it evades the explicit constructor
check right now, since the check only knows about the inline keyword.
Teach it that constexpr can be used also.

danakj committed:
Teach the explicit constructor check about constexpr.

If a constructor is marked constexpr it evades the explicit constructor
check right now, since the check only knows about the inline keyword.
Teach it that constexpr can be used also.

eglaysher commented:
Review: this looks fine, but please add a unit test. There's already a test testExplicitSingleArgumentConstructors(); just add a case in there to show that constexpr works.

danakj committed:
Add unit tests for constexpr constructors

This also adds unit tests for inline+explicit constructors being okay,
with and without constexpr.

danakj commented:
Thanks, I've added a bunch of tests, please have a look.

eglaysher commented:
LG. Thank you for being thorough on the tests.
